auslib/db.py

```diff
@@ -946,7 +946,27 @@ def getReleaseInfo(self, product=None, limit=None,
             column = [self.name]
         else:
             column = [self.name, self.product, self.data_version, self.read_only]
+
         rows = self.select(where=where, columns=column, limit=limit, transaction=transaction)
+
+        if not nameOnly:
+            j = join(dbo.releases.t, dbo.rules.t, ((dbo.releases.name == dbo.rules.mapping) | (dbo.releases.name == dbo.rules.whitelist)))
+            ref_list = select([dbo.releases.name, dbo.rules.rule_id]).select_from(j).execute().fetchall()
+            ref_dict = {}
+
+            for ref in ref_list:
+                rel_name, rule_id = ref
+                try:
+                    ref_dict[rel_name].append(rule_id)
+                except KeyError:
+                    ref_dict[rel_name] = [rule_id]
+
+            for row in rows:
+                try:
+                    row['rule_ids'] = ref_dict[row['name']]
+                except KeyError:
+                    row['rule_ids'] = []
```

Comment:
I think the logic of these two loops can be improved. If you nest them, you should be able to get rid of the "ref_dict" intermediary structure. I also think using ifs instead of try/except would make this read better. Eg:
if "rule_ids" not in row:
  row["rule_ids"] = ...
else:
  row["rule_ids"] = []